flow/sequence.go

```diff
@@ -0,0 +1,16 @@
+package flow
+
+import "time"
+
+// Sequence is a list of states that will be run in the order provided.
+type Sequence []State
+
+// Progress represents the progress of a currently running Sequence.
+type Progress struct {
+	CurrentState  State
+	StateIndex    int
+	Complete      bool
+	Sequence      Sequence
```

Comment:
This is so that a dispatcher knows the overall running sequence. The Sequencer itself though does not need to have sequence so I did remove that. The progress object in sequencer is purely for publishing to its subscribers.